python_ta/reporters/node_printers.py

```diff
@@ -417,40 +423,45 @@ def render_pep8_errors_e302(msg, line, col, source_line=None):
         )
     else:
         line -= 1
+        yield from render_context(line - 1, line + 1, source_lines)
         yield from render_blank_line(line)
         yield (None, slice(None, None), LineType.ERROR, NEW_BLANK_LINE_MESSAGE)
+    yield from render_context(msg.line, msg.line + 2, source_lines)
 
 
-def render_pep8_errors_e303(msg, line, col, source_line=None):
+def render_pep8_errors_e303(msg, source_lines=None):
     """Render a PEP8 too many blank lines message."""
-    line -= 1
-    while source_line.strip() == "":
+    line = msg.line - 1
+    while source_lines[line - 1].strip() == "":
         line -= 1
-
-    body = source_line[msg.line - 1]
+    yield from render_context(line - 1, line + 1, source_lines)
+    body = source_lines[msg.line - 1]
     indentation = len(body) - len(body.lstrip())
     yield from (
         (curr_line, slice(None, None), LineType.ERROR, " " * (indentation + 28))
         for curr_line in range(line + 1, msg.line)
     )
+    yield from render_context(msg.line, msg.line + 2, source_lines)
 
 
-def render_pep8_errors_e304(msg, line, col, source_line=None):
+def render_pep8_errors_e304(msg, source_lines=None):
```

Comment:
I think you can just combine this function with E303. The only difference is the presence of `indentation` in E303, which is actually good to do for E304 as well.